src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildPipelineNode.java

```diff
@@ -0,0 +1,338 @@
+package org.datadog.jenkins.plugins.datadog.model;
+
+import hudson.console.AnnotatedLargeText;
+import hudson.model.Node;
+import hudson.model.Result;
+import org.datadog.jenkins.plugins.datadog.DatadogUtilities;
+import org.datadog.jenkins.plugins.datadog.traces.StepDataManager;
+import org.jenkinsci.plugins.workflow.actions.ArgumentsAction;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepAtomNode;
+import org.jenkinsci.plugins.workflow.graph.BlockEndNode;
+import org.jenkinsci.plugins.workflow.graph.BlockStartNode;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.graph.StepNode;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represent a stage of the Jenkins Pipeline.
+ */
+public class BuildPipelineNode {
+
+    public enum NodeType {
+        PIPELINE("pipeline"), STAGE("stage"), STEP("step", "job");
+
+        private final String name;
+        private final String normalizedName;
+
+        NodeType(final String name) {
+            this.name = name;
+            this.normalizedName = name;
+        }
+
+        NodeType(final String name, final String normalizedName) {
+            this.name = name;
+            this.normalizedName = normalizedName;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getNormalizedName() {
+            return normalizedName;
+        }
+    }
+
+    private final BuildPipelineNodeKey key;
+    private final List<BuildPipelineNode> children;
+    private final String id;
+    private final String name;
+
+    private NodeType type;
+    private boolean internal;
+    private Map<String, Object> args = new HashMap<>();
+    private Map<String, String> envVars = new HashMap<>();
+    private String workspace;
+    private String nodeName;
+    private String nodeHostname;
+    private AnnotatedLargeText logText;
+    private long startTime;
+    private long startTimeMicros;
+    private long endTime;
+    private long endTimeMicros;
+    private String result;
+    private boolean error;
+    private Throwable errorObj;
```

Comment:
Same comment as above but with `error` and `errorObj`

Reply:
In this context, the `error` flag can be true but `errorObj` can be null when the status `ERROR` is propagated to the parent nodes.

Added some comments to bring more light to this fact.